### ISSUE TITLE
Mesh sharpness decreased for the latest code

### ISSUE THREAD

**Issue by krahets** [autonomousvision/sdfstudio#54]
Hi @niujinshuchong ! I found that the latest codes produce meshes with less sharpness compared to the codes before https://github.com/autonomousvision/sdfstudio/commit/6f51ffdd861629150d4a91a573347d35d7f61618 . 

- Only tested neus-facto.
- All the args related to RefNeRF were turned off for consistency. 

I've read the code changes and it's still not clear what the problem is. Were you got the same result?

**Comment by niujinshuchong** [comment]
@krahets Thanks for pointing this out. It might be because now we normalize the input points by 4 (https://github.com/autonomousvision/sdfstudio/blob/master/nerfstudio/fields/sdf_field.py#L357) instead of 2 in the previous commit. So only 1/8 of the grid features are used. I will make an update soon.

**Comment by krahets** [comment]
I also tested modifying `positions = (inputs + 1.0) / 2.0` to `positions = (inputs + 2.0) / 4.0` based on the code https://github.com/autonomousvision/sdfstudio/commit/a83e5623cda8b4f95abbcee1a5faeb1f8038ce94. And the sharpness of the meshes is the same.

It seems like `positions = (inputs + 2.0) / 4.0` normalized the contracted space from `[-2, 2]` to `[0, 1]` while `positions = (inputs + 1.0) / 2.0` mapped to `[-0.5, 1.5]`. I'm curious about whether hash encoding will be affected by the different normalization ranges.

**Comment by niujinshuchong** [comment]
@krahets In the previous version, the contraction is not used because we use a background model when we need to model complex background. Could you share some of your results? and also the training commands.

**Comment by krahets** [comment]
I found the scene contraction in the previous version https://github.com/autonomousvision/sdfstudio/tree/a83e5623cda8b4f95abbcee1a5faeb1f8038ce94

- https://github.com/autonomousvision/sdfstudio/blob/a83e5623cd/nerfstudio/models/base_surface_model.py#L142
- https://github.com/autonomousvision/sdfstudio/blob/a83e5623cd/nerfstudio/fields/sdf_field.py#L322

I think the scene contraction is used in this version. If we use near far collider and set the `near = 0.01` and `far = 2.0`, the range `(1.0, 2.0]` will be distorted by this contraction. I'm wondering whether I make mistakes on this.

> Could you share some of your results? and also the training commands.

Sure.

**Comment by krahets** [comment]
It seems like the high-frequency details degrade.

Previous code:
<img width="477" alt="image" src="https://user-images.githubusercontent.com/26993056/224310468-67789c8c-ecaf-409a-bf4b-0622185ca10d.png">

Latest code:
<img width="477" alt="image" src="https://user-images.githubusercontent.com/26993056/224310522-4b739c6a-7234-4b2a-b9dd-926929240741.png">

**Comment by niujinshuchong** [comment]
@krahets Thanks for sharing the results. There are four minor differences in the code.
1. points normalisation: `positions = (inputs + 2.0) / 4.0` vs `positions = (inputs + 1.0) / 2.0`
2. gradient normalisation: we don't normalise the gradient in the previous version https://github.com/autonomousvision/sdfstudio/blob/master/nerfstudio/fields/sdf_field.py#L477
3. Initialisations for the color mlp: https://github.com/autonomousvision/sdfstudio/blob/master/nerfstudio/fields/sdf_field.py#L336-L337
4. Rgb padding from mipnerf.

Could you try the other three to check whether it causes the problem. Thanks.

**Comment by krahets** [comment]
Tested them all and the reason is `2.`. Fix it at #56

I think it is because the gradients provide more information than the normals. Looking forward to your insights.

**Comment by niujinshuchong** [comment]
@krahets Thanks for sharing the results. It's a very interesting finding. I guess it might because there is one more degree of freedom in gradients so that the color network can change the geometry more easily to minimise RGB loss.
